go/database/mpt/nodes.go

```diff
@@ -473,10 +478,15 @@ func (c *nodeCheckContext) isCompatible(other *nodeCheckContext) bool {
 type nodeBase struct {
 	hash       common.Hash // the hash of this node (may be dirty)
 	hashStatus hashStatus  // indicating whether this node's hash is valid
-	clean      bool        // by default nodes are dirty (clean == false)
+	clean      int32       // by default nodes are dirty
```

Comment:
why not using `atomic.Bool`

Reply:
I was initially using an atomic Bool, but this caused multiple issues:
- `atomic.Bool` should not be copied and/or compared; however, we do so explicitly or implicitly in many situations; if this flag is switched to an `atomic.Bool`, `go vet` is reporting thousands of issues in our test code
- `atomic.Bool` can not be inline-initialized when defining structs; something like `ValueNode{base: nodeBase{clean: false}}` would have to be rewritten into two statements, one initializing the node and another explicitly marking it as dirty. This would touch a huge number of code locations

I switched between the `atomic.Bool` and this version several times, but in the end went for this one as the lesser of two evils.

Reply:
ok, thanks for explanation